etcdutl/snapshot/v3_snapshot.go

```diff
@@ -265,6 +274,13 @@ func (s *v3Manager) Restore(cfg RestoreConfig) error {
 	if err = s.saveDB(); err != nil {
 		return err
 	}
+
+	if cfg.MarkCompacted || cfg.RevisionBump > 0 {
```

Comment:
I'm not sure that's necessarily the case; we need to compact if we're bumping the revision, but if we're not bumping the revision, the user may still want to mark everything in the restore as compacted. That was at least my understanding as to why to have them be separate options. @serathius thoughts?

Reply:
> but if we're not bumping the revision, the user may still want to mark everything in the restore as compacted

Why? What this PR does is already a little hacky, why do we need to provide more flexibility? I do not see any real use case, but it provides flexibility for users to make mistake.

Reply:
Separate flags for clarity, but I agree with @ahrtr that we don't need to make them independent. It would only require more work to test all edge cases.